code/modules/unit_tests/organs.dm

```diff
@@ -0,0 +1,174 @@
+		var/list/obj/item/organ/inserted_organs = list()
+		// Attempt to insert entire list of mutant organs for the given infusion_entry.
+		for(var/obj/item/organ/organ as anything in output_organs)
+			organ = new organ()
+			var/implant_ok = organ.Insert(lab_rat, special = TRUE, drop_if_replaced = FALSE)
+			if(!implant_ok)
+				TEST_FAIL("The organ \"[organ.type]\" for \"[infuser_entry.type]\" was not inserted in the mob when expected.")
+				continue
+			inserted_organs += organ
+
+		// Search for added Status Effect.
+		var/datum/status_effect/organ_set_bonus/added_status = check_status_type(lab_rat, /datum/status_effect/organ_set_bonus)
+
+		// Threshold description implies an organ set bonus.
+		// Without it, we'll assume there isn't a Status Effect to look for.
+		var/has_threshold = (infuser_entry.threshold_desc != DNA_INFUSION_NO_THRESHOLD)
+		// How many organs the Status Effect requires to be inserted before it will activate.
+		var/total_organs_needed = added_status ? added_status.organs_needed : 0
+		// How many organs are available from the infuser entry.
+		var/total_organs = length(infuser_entry.output_organs)
+		// Quantity of successfully inserted organs.
+		var/total_inserted = length(inserted_organs)
+
+		// Found a Status Effect but no threshold description.
+		if(!has_threshold && added_status)
+			TEST_FAIL("The threshold_desc variable for \"[infuser_entry.type]\" was an empty string when a description was expected.")
+
+		// Since threshold_desc is filled-in, we expect the organ set bonus.
+		// Each of these fails in-order, so your test output doesn't get spammed with messages.
+		if(has_threshold)
+			if(!added_status)
+				TEST_FAIL("The \"/datum/status_effect/organ_set_bonus\" for \"[infuser_entry.type]\" was not added to the mob when expected.")
+			else if(!total_organs_needed)
+				TEST_FAIL("The \"needed_organs\" variable for \"[added_status.type]\" was 0 or null, when a positive number was expected.")
+			else if(total_organs_needed > total_organs)
+				TEST_FAIL("The \"output_organs\" list for \"[infuser_entry.type]\" had a length of \"[length(infuser_entry.output_organs)]\" when a minimum of at least [total_organs_needed] organs was specified in \"[added_status.type]\".")
+			else if(!added_status.bonus_active)
+				TEST_FAIL("The \"[added_status.type]\" bonus was not activated after inserting [total_inserted] of the [total_organs_needed] required organs in the mob, when it was expected.")
+
+		// Nothing to do.
+		if(!total_inserted)
+			continue
+
+		// Bonus of the Fly mutation swaps out all the organs, making it rather permanent.
+		// As a result, the inserted_organs list is un-usable by this point.
+		if(istype(infuser_entry, /datum/infuser_entry/fly))
+			continue
+
+		// Remove all the organs which were just added.
+		var/total_removed = 0
+		for(var/obj/item/organ/test_organ as anything in inserted_organs)
+			test_organ.Remove(lab_rat, special = TRUE)
+			total_removed += 1
```

Comment:
What's the point of incrementing this variable unconditionally instead of `inserted_organs.len`?